### PR TITLE
Add new rules visualizing CARLISLE results

### PR DESCRIPTION
**Pull request by epehrsson**
## Changes

Adds rules:
- count_peaks - output table with number of peaks per condition, visualize number of peaks and correlation between dedup and no_dedup peaks
- cov_correlation - find Pearson correlation in genome-wide coverage for all samples, plot heatmap and PCA
- homer_enrich - visualize HOMER enrichment over genic features for each condition
- combine_homer - add MACS2 fold change and q-value to HOMER annotation table 

## PR Checklist

- [x] This comment contains a description of changes with justifications, with any relevant issues linked.
- [x] Update docs if there are any API changes. - none
- [x] Update `CHANGELOG.md` with a short description of any user-facing changes and reference the PR number. Guidelines: https://keepachangelog.com/en/1.1.0/

### CODE REVIEW

**Comment by kelly-sovacool** [comment]
Thanks for opening this PR, @epehrsson! Sorry I didn't notice it -- in the future you can tag me with `@kelly-sovacool` somewhere in the this thread to notify me if you want me to review it.

Were you able to run this successfully on biowulf?

**Comment by epehrsson** [comment]
Hi @kelly-sovacool, sorry for not tagging you! Yes, I could run it on the test data and on two projects I'd processed earlier, which recapitulated the results of the rules. =

**Comment by kelly-sovacool** [comment]
Thanks for this contribution @epehrsson! I proposed changes [here](https://github.com/CCRGeneticsBranch/CARLISLE/pull/1) so that your new rules will also use the docker container. Please take a look and merge that if it looks good, then we'll run one more test to make sure I didn't break anything before merging this PR into main.

**Comment by epehrsson** [comment]
Thanks, Kelly! The changes look good, and I completed the pull request.

File: workflow/rules/annotations.smk [hunk 1]
```diff
@@ -60,6 +60,42 @@ rule findMotif:
         fi
         """
 
+rule homer_enrich:
+    """
+    Plot enrichment over genic features
+    """
+    input: 
+        annotation_summary=expand(join(RESULTSDIR,"peaks","{{qthresholds}}","{{peak_caller}}","annotation","homer","{treatment_control_list}.{{dupstatus}}.{{peak_caller_type}}.annotation.summary"),treatment_control_list=TREATMENT_LIST_M)
```
Comment (kelly-sovacool) [inline workflow/rules/annotations.smk:68]:
```suggestion

def get_annotation_files(wildcards):
    """
    treatment_control_list depends on the peak caller
    """
    return expand(join(RESULTSDIR,"peaks", wildcards.qthresholds, wildcards.peak_caller, "annotation","homer",
           "{treatment_control_list}" + "." + wildcards.dupstatus + "." + wildcards.peak_caller_type + ".annotation.summary"),
           treatment_control_list = TREATMENT_LIST_M if wildcards.peak_caller.startswith('macs2') else TREATMENT_LIST_SG)
    

rule homer_enrich:
    """
    Plot enrichment over genic features
    """
    input: 
        annotation_summary=get_annotation_files
```

**Review by kelly-sovacool — CHANGES_REQUESTED** [review]

**Pull request merged**